.github/workflows/go-tests.yml

```diff
@@ -0,0 +1,124 @@
+---
+# Ci code for building release artifacts.
+name: Go Tests
+on:
+  push: # Master branch checks only validate the build and generate artifacts for testing.
+    branches:
+      - master
+  pull_request: null # PR checks only validate the build and generate artifacts for testing.
+concurrency: # This keeps multiple instances of the job from running concurrently for the same ref and event type.
+  group: go-test-${{ github.ref }}-${{ github.event_name }}
+  cancel-in-progress: true
+jobs:
+  file-check: # Check what files changed if we’re being run in a PR or on a push.
+    name: Check Modified Files
+    runs-on: ubuntu-latest
+    outputs:
+      run: ${{ steps.check-run.outputs.run }}
+    steps:
+      - name: Checkout
+        id: checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          submodules: recursive
+      - name: Check files
+        id: check-files
+        uses: tj-actions/changed-files@v42
+        with:
+          since_last_remote_commit: ${{ github.event_name != 'pull_request' }}
+          files: |
+            **/*.cmake
+            CMakeLists.txt
+            .github/workflows/go-tests.yml
+            packaging/cmake/
+            src/go/**
+          files_ignore: |
+            **/*.md
+            src/go/**/metadata.yaml
+      - name: List all changed files in pattern
+        continue-on-error: true
+        env:
+          ALL_CHANGED_FILES: ${{ steps.check-files.outputs.all_changed_files }}
+        run: |
+          for file in ${ALL_CHANGED_FILES}; do
+            echo "$file was changed"
+          done
+      - name: Check Run
+        id: check-run
+        run: |
+          if [ "${{ steps.check-files.outputs.any_modified }}" == "true" ] || [ "${{ github.event_name }}" == "workflow_dispatch" ]; then
+            echo 'run=true' >> "${GITHUB_OUTPUT}"
+          else
+            echo 'run=false' >> "${GITHUB_OUTPUT}"
+          fi
+
+  matrix:
+    name: Generate Build Matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.get-version.outputs.matrix }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Install dependencies
+        run: sudo apt-get update && sudo apt-get upgrade -y && sudo apt-get install -y python3-packaging
+      - name: Get Go version and modules
+        id: get-version
+        run: .github/scripts/get-go-version.py
```

Comment:
We never skip this job, why?

Reply:
Put differently, this is the same reason that we always generate the build matrix for the build, docker, and packaging CI workflows.